yt/frontends/amrvac/data_structures.py

```diff
@@ -259,7 +259,7 @@ def _set_code_unit_attributes(self):
 
         He_abundance = 0.1  # hardcoded parameter in AMRVAC
         density_unit = (1.0 + 4.0*He_abundance) * mass_hydrogen_cgs * numberdensity_unit
-        if velocity_unit == 0:
+        if velocity_unit.value == 0:
```

Comment:
Yes, so that's how it's implemented in AMRVAC. The default value of `velocity_unit` is 0 and `temperature_unit` is 1, because then it normalises based on length, number density and temperature (by default).

That's also what's happening here: if `temperature_unit` is specified the other normalisations are calculated based on this (including the temperature). If `velocity_unit` is specified the normalisation uses that instead of temperature.

You always have only 3 degrees of freedom: `lenght_unit`, `numberdensity_unit` you always have, and then EITHER `velocity_unit` or `temperature_unit`.

Reply:
Oh, I see. Does it mean that we should throw an error in case the user tries to override both velocity and temperature units ?

Reply:
It's not necessarily a bad thing to "override" the override method. It's call subclassing and it *may* actually be what you are after. I recently simplified our frontend to reduce the amount of methods we were subclassing, but if there is a need for more, it's fine to do it. That being said, we should first lookup other frontends to see if anyone else is doing it, otherwise it's probably a bad idea.